### PR TITLE
fix: token balance

### PR DESCRIPTION
**Pull request by alaibe**

### CODE REVIEW

**Review by stefandunca — APPROVED** [review]
Great finding!

**Review by IvanBelyakoff — APPROVED** [review]

**Review by dlipicar — APPROVED** [review]

**Review by rasom — APPROVED** [review]
My fault

File: services/wallet/token/token.go [hunk 1]
```diff
@@ -783,7 +783,10 @@ func (tm *Manager) GetBalancesAtByChain(parent context.Context, clients map[uint
 		mu.Unlock()
 	}
 
-	for _, client := range clients {
+	for clientIdx := range clients {
+		// Keep the reference to the client. DO NOT USE A LOOP, the client will be overrident in the coroutine
```
Comment (rasom) [inline services/wallet/token/token.go]: overriden

Comment (rasom) [reply services/wallet/token/token.go]: same for other comments

**Comment by rasom** [comment]
Just wondering, is there a safer way to achieve same results (not like I fully understand the reasoning here), or that's an idiomatic way to handle this referencing issue?

**Comment by alaibe** [comment]
@rasom from what I know there is no other way. When there is a coroutine, you need assign variable otherwise they will be rewrite.
That's a bit naughty but according to golang this is an optimization

@stefandunca shared this with me: https://golang50shad.es/index.html#range_val_update which may explain the problem better than i would ever do

**Pull request merged**